### PR TITLE
fix: use the custom image factory host in backend and frontend

### PR DESCRIPTION
**Pull request by utkuozdemir**
Contains two fixes:

- In the frontend, `factory.talos.dev` was hardcoded for the maintenance upgrade calls. Move the image factory URL into the `FeaturesConfig` resource, and switch to reading the host from this resource on the frontend.

- In the backend, we were initializing the `imageFactoryHost` in an `init` block, and because of init, it was running before the user override was actually written into the config by Cobra. Fix it by initializing the `imageFactoryHost` lazily when it is used for the first time.

Fixes siderolabs/omni#711.

### CODE REVIEW

File: internal/backend/runtime/omni/controllers/omni/cluster_machine_config.go [hunk 1]
```diff
@@ -432,16 +433,14 @@ func stripTalosAPIAccessOSAdminRole(cfg config.Provider) (config.Provider, error
 	return container.New(updatedDocs...)
 }
 
-var imageFactoryHost string
-
-func init() {
+var getImageFactoryHost = sync.OnceValues(func() (string, error) {
```
Comment (utkuozdemir) [inline internal/backend/runtime/omni/controllers/omni/cluster_machine_config.go:436]: Backend fix

File: frontend/src/methods/machine.ts [hunk 2]
```diff
@@ -151,8 +152,12 @@ const copyUserLabels = (src: Resource, dst: Resource) => {
 }
 
 export const updateTalosMaintenance = async (machine: string, talosVersion: string, schematic?: string) => {
+  const imageFactoryBaseURL = await getImageFactoryBaseURL();
+
+  const host = new URL(imageFactoryBaseURL).host;
+
   const image = schematic ?
-    `factory.talos.dev/installer/${schematic}:v${talosVersion}` :
+    `${host}/installer/${schematic}:v${talosVersion}` :
```
Comment (utkuozdemir) [inline frontend/src/methods/machine.ts:160]: Frontend fix

File: frontend/src/methods/machine.ts [hunk 3]
```diff
@@ -151,8 +152,12 @@ const copyUserLabels = (src: Resource, dst: Resource) => {
...
     `ghcr.io/siderolabs/installer:v${talosVersion}`;
```
Comment (frezbo) [inline frontend/src/methods/machine.ts:161]: i guess this is also hardcoded in a way :sweat_smile:

**Review by frezbo — APPROVED** [review]

**Review by smira — APPROVED** [review]

File: frontend/src/methods/machine.ts (hunk 3 above)
Comment (utkuozdemir) [reply frontend/src/methods/machine.ts:161]: Yep, but probably much less of a problem than the other. Can live with it for now I think.

**Pull request merged**